geode-core/src/main/java/org/apache/geode/admin/internal/ThreadMonitoringProcess.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.admin.internal;
+
+import java.util.Date;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.TimerTask;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.CacheClosedException;
+import org.apache.geode.distributed.internal.DistributionConfigImpl;
+import org.apache.geode.internal.cache.GemFireCacheImpl;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.cache.control.ResourceManagerStats;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.internal.statistics.AbstractExecutorGroup;
+
+
+public class ThreadMonitoringProcess extends TimerTask {
+
+  private ResourceManagerStats resourceManagerStats = null;
+  private static final Logger logger = LogService.getLogger();
+  private static int timeLimit;
+
+  Properties nonDefault = new Properties();
+  DistributionConfigImpl dcI = new DistributionConfigImpl(nonDefault);
+
+  protected ThreadMonitoringProcess() {
+    timeLimit = dcI.getThreadMonitorTimeLimit();
+  }
+
+  public boolean mapValidation() {
+    boolean isStuck = false;
+    int numOfStuck = 0;
+    long delta;
```

Comment:
I'm not sure why you're declaring variables up-front in a method like this. Looks like old style C instead of Java. You should just declare the variable where it's used. Also, don't try to out-smart the JVM and JIT by declaring vars outside a loop. The variable should be declared within the scope that it's used. So if the variable is set within the loop but accessed after the loop, then ok, declare if before the loop. But any variable that isn't accessed after the loop should just be declared inside the loop.